lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -590,4 +590,58 @@ def thermal_zone_prm_lab_delta_t(thermal_zone)
   def model_get_fan_power_breakdown
     return true
   end
+
+  def model_set_spm(model, thermalZones, coil)
+    # search for the highest zone setpoint temperature
+    max_heat_setpoint = 0.0
+    coil_name = coil.name.get.to_s
+    thermalZones.each do |zone|
+      tstat = zone.thermostatSetpointDualSetpoint
+      if tstat.is_initialized
+        tstat = tstat.get
+        setpoint_sch = tstat.heatingSetpointTemperatureSchedule
+        if setpoint_sch.is_initialized
+          setpoint_sch = setpoint_sch.get
+          if setpoint_sch.to_ScheduleRuleset.is_initialized
+            setpoint_sch = setpoint_sch.to_ScheduleRuleset.get
+            setpoint_c = schedule_ruleset_annual_min_max_value(setpoint_sch)['max']
+          elsif setpoint_sch.to_ScheduleConstant.is_initialized
+            setpoint_sch = setpoint_sch.to_ScheduleConstant.get
+            setpoint_c = schedule_constant_annual_min_max_value(setpoint_sch)['max']
+          elsif setpoint_sch.to_ScheduleCompact.is_initialized
+            setpoint_sch = setpoint_sch.to_ScheduleCompact.get
+            setpoint_c = schedule_compact_annual_min_max_value(setpoint_sch)['max']
+          end
+          if setpoint_c > max_heat_setpoint
+            max_heat_setpoint = setpoint_c
+          end
+        end
+      end
+    end
```

Comment:
The decision for such implementation was based on:
1. An assumption that winter design day setpoint shall be the max value of an annual heating setpoint schedule.
2. Existing functions available in the code base for search for finding the annual maximum heating setpoint.

It is may not straightforward but should work well with the above assumptions.

@dmaddoxwhite could you take a look at this comment?

Reply:
@lymereJ @dmaddoxwhite 
The new code is pushed to the repo for merge. The setpoint search is relocated to the Standard.Model.rb as a utility function.
Getting min max values from design day is implemented for both ruleset and compact.
Constant schedule is ignored in this update because the schedule has only one value.